feature-libs/checkout/base/core/facade/checkout-delivery-address.service.spec.ts

```
@@ -0,0 +1,227 @@
+  });
+
+  it(`should inject CheckoutDeliveryAddressService`, inject(
+    [CheckoutDeliveryAddressService],
+    (checkoutDeliveryAddressService: CheckoutDeliveryAddressService) => {
+      expect(checkoutDeliveryAddressService).toBeTruthy();
+    }
+  ));
+
+  describe(`getDeliveryAddress`, () => {
+    it(`should return the address`, (done) => {
+      spyOn(checkoutQuery, 'getCheckoutDetailsState').and.returnValue(
+        of(<QueryState<CheckoutState>>{
+          loading: false,
+          error: false,
+          data: {
+            deliveryAddress: mockAddress,
+          },
+        })
+      );
+
+      service
+        .getDeliveryAddressState()
+        .pipe(take(1))
+        .subscribe((result) => {
+          expect(result).toEqual(<QueryState<Address | undefined>>{
+            loading: false,
+            error: false,
+            data: mockAddress,
+          });
+          done();
+        });
+    });
+  });
+
+  describe(`createAndSetAddress`, () => {
+    it(`should call checkoutDeliveryConnector.createAddress`, () => {
+      spyOn(connector, 'createAddress').and.stub();
+
+      service.createAndSetAddress(mockAddress);
+
+      expect(connector.createAddress).toHaveBeenCalledWith(
+        mockUserId,
+        mockCartId,
+        mockAddress
+      );
+    });
+
+    it(`should call facade's setDeliveryAddress()`, () => {
+      spyOn(service, 'setDeliveryAddress').and.stub();
+
+      service.createAndSetAddress(mockAddress);
+
+      expect(service.setDeliveryAddress).toHaveBeenCalledWith(mockAddress);
+    });
+
+    // TODO: Replace with event testing once we remove ngrx store.
+    it(`should dispatch UserActions.LoadUserAddresses`, () => {
+      spyOn(store, 'dispatch').and.stub();
+
+      service.createAndSetAddress(mockAddress);
+
+      expect(store.dispatch).toHaveBeenCalledWith(
+        new UserActions.LoadUserAddresses(mockUserId)
+      );
+    });
+  });
+
+  describe(`setDeliveryAddress`, () => {
```

should we add a case that says `it(should throw error and NOT set/dispatch address, () => { when no address id is provided` for the branches coverage ;d

---

Done